src/Metadata/Resource/Factory/ResourceMetadataFactoryTrait.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace ApiPlatform\Core\Metadata\Resource\Factory;
+
+use ApiPlatform\Core\Exception\ResourceClassNotFoundException;
+use ApiPlatform\Core\Metadata\Resource\Operation;
```

Comment:
Yes, but by defining custom operations as arrays (through configurations or annotations), you've already defined everything needed to build a Route. Therefore, you won't go through `OperationResourceMetadataFactory` but you won't need any `Operation` or `Pagination`.

For example [this entity](https://github.com/api-platform/core/blob/master/tests/Fixtures/TestBundle/Entity/RelationEmbedder.php#L30) works fine, and doesn't go through the highlighted code path because it doesn't need to.

Check also [ApiLoader](https://github.com/api-platform/core/blob/master/src/Bridge/Symfony/Routing/ApiLoader.php#L92) where `$operation` is a simple `Array` with no `pagination` nor based on any `Operation` interface.